### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/pipeline_tutorial.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/pipeline_tutorial.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/pipeline_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+{'text': ' I have a dream that one day this nation will rise up and live out the true meaning of its creed.'}
+```
+
+الآن تبدو هذه النتيجة أكثر دقة! للحصول على مقارنة متعمقة حول Wav2Vec2 مقابل Whisper، راجع [دورة Audio Transformers](https://huggingface.co/learn/audio-course/chapter5/asr_models).
+نحن نشجعك بشدة على التحقق من Hub للحصول على نماذج بلغات مختلفة، ونماذج متخصصة في مجالك، وأكثر من ذلك.
+يمكنك التحقق من نتائج النموذج ومقارنتها مباشرة من متصفحك على Hub لمعرفة ما إذا كان يناسبها
+أو التعامل مع الحالات الحدية بشكل أفضل من غيرها.
+وإذا لم تجد نموذجًا لحالتك الاستخدام، فيمكنك دائمًا البدء في [التدريب](training) الخاص بك!
+
+إذا كان لديك عدة مدخلات، فيمكنك تمرير إدخالك كقائمة:
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:53]:
```suggestion

```
لوجود نفس الجمله مكرره

File: docs/source/ar/pipeline_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+دعونا نجربها هنا لنرى كيف تؤدي:
+
+```py
+>>> transcriber = pipeline(model="openai/whisper-large-v2")
+>>> transcriber("https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/mlk.flac")
+{'text': ' I have a dream that one day this nation will rise up and live out the true meaning of its creed.'}
+```
+
+الآن تبدو هذه النتيجة أكثر دقة! للحصول على مقارنة متعمقة حول Wav2Vec2 مقابل Whisper، راجع [دورة Audio Transformers](https://huggingface.co/learn/audio-course/chapter5/asr_models).
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:47]:
```suggestion
الآن تبدو هذه النتيجة أكثر دقة! لمقارنة عميقة حول Wav2Vec2 مقابل Whisper، راجع [دورة Audio Transformers](https://huggingface.co/learn/audio-course/chapter5/asr_models).
```

File: docs/source/ar/pipeline_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+نحن نشجعك بشدة على التحقق من Hub للحصول على نماذج بلغات مختلفة، ونماذج متخصصة في مجالك، وأكثر من ذلك.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:48]:
```suggestion
نشجعك بشدة على التحقق من Hub للحصول على نماذج بلغات مختلفة، ونماذج متخصصة في مجالك، وأكثر من ذلك.
```

File: docs/source/ar/pipeline_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+يمكنك التحقق من نتائج النموذج ومقارنتها مباشرة من متصفحك على Hub لمعرفة ما إذا كان يناسبها
+أو التعامل مع الحالات الحدية بشكل أفضل من غيرها.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:50]:
```suggestion
أو التعامل مع الحالات الخاصة بشكل أفضل من غيرها.
```

File: docs/source/ar/pipeline_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+وإذا لم تجد نموذجًا لحالتك الاستخدام، فيمكنك دائمًا البدء في [التدريب](training) الخاص بك!
+
+إذا كان لديك عدة مدخلات، فيمكنك تمرير إدخالك كقائمة:
+إذا كان لديك عدة مدخلات، فيمكنك تمرير إدخالك كقائمة:
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:54]:
```suggestion
إذا كان لديك عدة مدخلات، يمكنك تمرير مدخلك كقائمة:
```

File: docs/source/ar/pipeline_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+```py
+transcriber(
+    [
+        "https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/mlk.flac",
+        "https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/1.flac",
+    ]
+)
+```
+
+تعد خطوط الأنابيب رائعة للتجريب نظرًا لأن التبديل من نموذج إلى آخر أمر بسيط؛ ومع ذلك، هناك بعض الطرق لتحسينها لأحمال عمل أكبر من التجريب. راجع الأدلة التالية التي تتناول بالتفصيل التكرار عبر مجموعات البيانات الكاملة أو استخدام خطوط الأنابيب في خادم ويب:
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:65]:
```suggestion
تعد خطوط الأنابيب مثالية للتجريب نظرًا لأن التبديل من نموذج إلى آخر أمر بسيط للغاية؛ ومع ذلك، هناك بعض الطرق لتحسينها لأحمال عمل أكبر من التجريب. راجع الأدلة التالية التي تتعمق فى التكرار عبر مجموعات البيانات الكاملة أو استخدام خطوط الأنابيب في خادم ويب:
```

File: docs/source/ar/pipeline_tutorial.md [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+إذا كنت تستخدم `device=n`، فإن خط الأنابيب يضع النموذج تلقائيًا على الجهاز المحدد.
+سيعمل هذا بغض النظر عما إذا كنت تستخدم PyTorch أو Tensorflow.
+
+```py
+transcriber = pipeline(model="openai/whisper-large-v2", device=0)
+```
+
+إذا كان النموذج كبيرًا جدًا بالنسبة لوحدة معالجة الرسومات (GPU) واحدة، وأنت تستخدم PyTorch، فيمكنك تعيين `torch_dtype='float16'` لتمكين الاستدلال بدقة FP16. عادةً ما لا يتسبب ذلك في حدوث انخفاضات كبيرة في الأداء، ولكن تأكد من تقييمه على نماذجك!
+
+بدلاً من ذلك، يمكنك تعيين `device_map="auto"` لتحديد كيفية تحميل مخزنات النموذج وتخزينها تلقائيًا. يتطلب استخدام حجة `device_map` حزمة 🤗 [Accelerate](https://huggingface.co/docs/accelerate):
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:96]:
```suggestion
بدلاً من ذلك، يمكنك تعيين `device_map="auto"` لتحديد كيفية تحميل مخزنات النموذج وتخزينها تلقائيًا. يتطلب استخدام معامل `device_map` مكتبه 🤗 [Accelerate](https://huggingface.co/docs/accelerate):
```

File: docs/source/ar/pipeline_tutorial.md [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+
+```py
+transcriber = pipeline(model="openai/whisper-large-v2", device_map="auto")
+```
+
+لاحظ أنه إذا تم تمرير `device_map="auto"`، فلا توجد حاجة لإضافة حجة `device=device` عند إنشاء خط الأنابيب الخاص بك، فقد تواجه بعض السلوكيات غير المتوقعة!
+
+### حجم الدفعة
+
+بشكل افتراضي، لن تقوم خطوط الأنابيب بتجزئة الاستدلال لأسباب موضحة بالتفصيل [هنا](https://huggingface.co/docs/transformers/main_classes/pipelines#pipeline-batching). والسبب هو أن التجزئة ليست أسرع بالضرورة، ويمكن أن تكون أبطأ في الواقع في بعض الحالات.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:112]:
```suggestion
بشكل افتراضي، لن تقوم خطوط الأنابيب بتجميع الاستدلال لأسباب مفصلة [هنا](https://huggingface.co/docs/transformers/main_classes/pipelines#pipeline-batching). والسبب هو أن التجميع ليست أسرع بالضرورة، ويمكن أن تكون أبطأ في الواقع في بعض الحالات.
```

File: docs/source/ar/pipeline_tutorial.md [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+
+ولكن إذا نجحت في حالتك الاستخدام، فيمكنك استخدام ما يلي:
+
+```py
+transcriber = pipeline(model="openai/whisper-large-v2", device=0, batch_size=2)
+audio_filenames = [f"https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/{i}.flac" for i in range(1, 5)]
+texts = transcriber(audio_filenames)
+```
+
+هذا يشغل خط الأنابيب على ملفات الصوت الأربعة المقدمة، ولكنه سيمررها في دفعات من 2
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:122]:
```suggestion
هذا يشغل خط الأنابيب على ملفات الصوت الأربعة المتاحة، ولكنه سيمررها على دفعتين
```

File: docs/source/ar/pipeline_tutorial.md [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+إلى النموذج (الذي يوجد على وحدة معالجة الرسومات (GPU)، حيث من المرجح أن تساعد التجزئة) دون الحاجة إلى أي رمز إضافي منك.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:123]:
```suggestion
إلى النموذج (الذي يوجد على وحدة معالجة الرسومات (GPU)، حيث من المرجح أن تساعد التجميع) دون الحاجة إلى أي رمز إضافي منك. 
```

File: docs/source/ar/pipeline_tutorial.md [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+
+```py
+transcriber = pipeline(model="openai/whisper-large-v2", device=0, batch_size=2)
+audio_filenames = [f"https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/{i}.flac" for i in range(1, 5)]
+texts = transcriber(audio_filenames)
+```
+
+هذا يشغل خط الأنابيب على ملفات الصوت الأربعة المقدمة، ولكنه سيمررها في دفعات من 2
+إلى النموذج (الذي يوجد على وحدة معالجة الرسومات (GPU)، حيث من المرجح أن تساعد التجزئة) دون الحاجة إلى أي رمز إضافي منك. 
+يجب أن تتطابق الإخراج دائمًا مع ما كنت ستحصل عليه دون التجزئة. المقصود منه فقط كطريقة لمساعدتك في الحصول على مزيد من السرعة من خط الأنابيب.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:124]:
```suggestion
يجب أن تتطابق الإخراج دائمًا مع ما كنت ستحصل عليه دون التجميع. المقصود منه فقط كطريقة لمساعدتك في الحصول على سرعة أكبر من خط الأنابيب.
```

File: docs/source/ar/pipeline_tutorial.md [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+
+يمكن لخطوط الأنابيب أيضًا تخفيف بعض تعقيدات التجزئة لأنه، بالنسبة لبعض خطوط الأنابيب، يجب تقسيم عنصر واحد (مثل ملف صوتي طويل) إلى أجزاء متعددة لمعالجته بواسطة نموذج. يقوم خط الأنابيب بأداء هذه التجزئة [*batch batching*](./main_classes/pipelines#pipeline-chunk-batching) نيابة عنك.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:126]:
```suggestion
يمكن لخطوط الأنابيب أيضًا تخفيف بعض تعقيدات التجميع لأنه، بالنسبة لبعض خطوط الأنابيب، يجب تقسيم عنصر واحد (مثل ملف صوتي طويل) إلى أجزاء متعددة لمعالجته بواسطة نموذج. يقوم خط الأنابيب بأداء هذه  العملية التي تسمى تجميع الأجزاء [*batch batching*](./main_classes/pipelines#pipeline-chunk-batching) نيابة عنك.
```

File: docs/source/ar/pipeline_tutorial.md [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+
+توفر جميع المهام معلمات خاصة بالمهمة تتيح المرونة والخيارات الإضافية لمساعدتك في أداء عملك.
+على سبيل المثال، تحتوي طريقة [`transformers.AutomaticSpeechRecognitionPipeline.__call__`] على معلمة `return_timestamps` التي تبدو واعدة لترجمة مقاطع الفيديو:
+```py
+>>> transcriber = pipeline(model="openai/whisper-large-v2", return_timestamps=True)
+>>> transcriber("https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/mlk.flac")
+{'text': ' I have a dream that one day this nation will rise up and live out the true meaning of its creed.', 'chunks': [{'timestamp': (0.0, 11.88), 'text': ' I have a dream that one day this nation will rise up and live out the true meaning of its'}, {'timestamp': (11.88, 12.38), 'text': ' creed.'}]}
+```
+
+كما ترون، استنتج النموذج النص وأخرج أيضًا **متى** تم نطق الجمل المختلفة.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:138]:
```suggestion
كما ترون، استنتج النموذج النص.وكذلك حدد **وقت** نطق الجمل المختلفة.
```

File: docs/source/ar/pipeline_tutorial.md [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+
+تتوفر العديد من المعلمات لكل مهمة، لذا تحقق من مرجع API لكل مهمة لمعرفة ما يمكنك العبث به!
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:140]:
```suggestion
تتوفر العديد من المعلمات لكل مهمة، لذا تحقق من مرجع API لكل مهمة لمعرفة ما يمكنك تعديله!
```

File: docs/source/ar/pipeline_tutorial.md [hunk 15]
```diff
@@ -0,0 +1,316 @@
...
+```
+
+إذا لم تتمكن من العثور على معلمة قد تساعدك حقًا، فلا تتردد في [طلبها](https://github.com/huggingface/transformers/issues/new?assignees=&labels=feature&template=feature-request.yml)!
+
+
+## استخدام خطوط الأنابيب على مجموعة بيانات
+
+يمكن أيضًا تشغيل خط الأنابيب للاستدلال على مجموعة بيانات كبيرة. أسهل طريقة نوصي بها للقيام بذلك هي باستخدام مؤشر:
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:155]:
```suggestion
يمكن أيضًا تشغيل خط الأنابيب للاستدلال على مجموعة بيانات كبيرة. أسهل طريقة نوصي بها للقيام بذلك هي باستخدام المتكرر (iterator).:
```

File: docs/source/ar/pipeline_tutorial.md [hunk 16]
```diff
@@ -0,0 +1,316 @@
...
+        yield f"My example {i}"
+
+
+pipe = pipeline(model="openai-community/gpt2", device=0)
+generated_characters = 0
+for out in pipe(data()):
+    generated_characters += len(out[0]["generated_text"])
+```
+
+يقوم المؤشر `data()` بإرجاع كل نتيجة، ويعترف خط الأنابيب تلقائيًا
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:169]:
```suggestion
يقوم المؤشر `data()` بإرجاع كل نتيجة، ويتعرف خط الأنابيب تلقائيًا
```

File: docs/source/ar/pipeline_tutorial.md [hunk 17]
```diff
@@ -0,0 +1,316 @@
...
+الإدخال قابل للتحديد ويبدأ في جلب البيانات أثناء
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:170]:
```suggestion
المدخل قابل للتحديد ويبدأ في جلب البيانات أثناء
```

File: docs/source/ar/pipeline_tutorial.md [hunk 18]
```diff
@@ -0,0 +1,316 @@
...
+يستمر في معالجتها على وحدة معالجة الرسومات (GPU) (يستخدم هذا [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) تحت الغطاء).
+هذا أمر مهم لأنك لا تحتاج إلى تخصيص ذاكرة لمجموعة البيانات بأكملها
+ويمكنك تغذية وحدة معالجة الرسومات (GPU) بأسرع ما يمكن.
+
+نظرًا لأن التجزئة قد تسرع الأمور، فقد يكون من المفيد ضبط معلمة `batch_size` هنا.
```
Comment (abodacs) [inline docs/source/ar/pipeline_tutorial.md:175]:
```suggestion
نظرًا لأن التجميع قد تسرع الأمور، فقد يكون من المفيد ضبط معلمة `batch_size` هنا.
```

**Review by abodacs — CHANGES_REQUESTED** [review]
سﻻم عليكم
شكرا لمجهودك @AhmedAlmaghz 

تم اضافة بعض المﻻحظات

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs

**Pull request closed without merging**